onnxruntime/python/tools/transformers/benchmark.py

```diff
@@ -96,13 +101,23 @@ def load_pretrained_model(model_name, config, cache_dir):
     return AutoModel.from_pretrained(model_name, config=config, cache_dir=cache_dir)
 
 
-def create_onnxruntime_session(onnx_model_path, use_gpu, enable_all_optimization):
+class Precision(Enum):
+    FLOAT32 = 'fp32'
+    FLOAT16 = 'fp16'
+    INT8 = 'int8'
+
+    def __str__(self):
+        return self.value
+
+
+def create_onnxruntime_session(onnx_model_path, use_gpu, enable_all_optimization=True, thread_num=-1):
     import onnxruntime
     sess_options = onnxruntime.SessionOptions()
     sess_options.graph_optimization_level = onnxruntime.GraphOptimizationLevel.ORT_ENABLE_ALL
     if not enable_all_optimization:
         sess_options.graph_optimization_level = onnxruntime.GraphOptimizationLevel.ORT_ENABLE_BASIC
 
+    sess_options.intra_op_num_threads = thread_num
```

Comment:
This does not have impact for onnxruntime cpu package. Consider update os.environ["OMP_NUM_THREADS"] as well.

Reply:
this one is for ORT package that is built without openmp. The openmp setting is set through torch.set_num_thread

Reply:
I think this will confuse user on two points: onnxruntime uses pytorch to set thread number; sess_options.intra_op_num_threads does not work for cpu package and no comment here.

I suggest we add torch.set_num_thread to PyTorch benchmark code path only, and OMP_NUM_THREADS to onnxruntime cpu benchmark code path (right before importing onnxruntime).